pkg/cmd/cli/cmd/set/triggers.go

```diff
@@ -656,6 +656,11 @@ func (t *TriggerDefinition) Apply(obj runtime.Object) error {
 			allNames.Insert(container.Name)
 		}
 		for _, trigger := range t.ImageChange {
+			// If we have only one container, assume the triggers should be set for that
+			// container and do not require --containers to be specified.
+			if len(allNames) == 1 {
+				trigger.Names = append(trigger.Names, allNames.List()[0])
+			}
 			if len(trigger.Names) == 0 {
 				return fmt.Errorf("you must specify --containers when setting --from-image")
```

Comment:
The error message must contain that `-c` is required when multiple containers exist in a template.
"you must specify --containers when using --from-image with multiple containers"